src/libtriton/arch/irBuilder.cpp

```diff
@@ -162,72 +171,26 @@ namespace triton {
        * concrete expressions and their AST nodes.
        */
       if (this->symbolicEngine->isEnabled() && this->modes.isModeEnabled(triton::modes::ONLY_ON_SYMBOLIZED)) {
-        /* Clean memory operands */
-        for (auto it = inst.operands.begin(); it!= inst.operands.end(); it++) {
-          if (it->getType() == triton::arch::OP_MEM) {
-            if (it->getMemory().getLeaAst()->isSymbolized() == false) {
-              this->astGarbageCollector.extractUniqueAstNodes(uniqueNodes, it->getMemory().getLeaAst()); // FIXME: without this line -> win32 failed
-              // FIXME: Should extract nodes
-              it->getMemory().setLeaAst(nullptr);
-            }
-          }
-        }
+        /* Clear memory operands */
+        this->collectUnsymbolizedNodes(uniqueNodes, inst.operands);
 
-        /* Clean implicit and explicit semantics - MEM */
-        for (auto it = loadAccess.begin(); it != loadAccess.end();) {
-          if (std::get<1>(*it)->isSymbolized() == false)
-            // FIXME: Should extract nodes
-            loadAccess.erase(it++);
-          else
-            ++it;
-        }
+        /* Clear implicit and explicit semantics - MEM */
+        this->collectUnsymbolizedNodes(loadAccess);
 
-        /* Clean implicit and explicit semantics - REG */
-        for (auto it = readRegisters.begin(); it != readRegisters.end();) {
-          if (std::get<1>(*it)->isSymbolized() == false)
-            // FIXME: Should extract nodes
-            readRegisters.erase(it++);
-          else
-            ++it;
-        }
-
-        /* Clean implicit and explicit semantics - IMM */
-        for (auto it = readImmediates.begin(); it != readImmediates.end();) {
-          if (std::get<1>(*it)->isSymbolized() == false)
-            // FIXME: Should extract nodes
-            readImmediates.erase(it++);
-          else
-            ++it;
-        }
+        /* Clear implicit and explicit semantics - REG */
+        this->collectUnsymbolizedNodes(readRegisters);
 
-        /* Clean implicit and explicit semantics - MEM */
-        for (auto it = storeAccess.begin(); it != storeAccess.end();) {
-          if (std::get<0>(*it).getLeaAst() == nullptr && std::get<1>(*it)->isSymbolized())
-            ++it;
+        /* Clear implicit and explicit semantics - IMM */
+        this->collectUnsymbolizedNodes(readImmediates);
 
-          else if (std::get<0>(*it).getLeaAst() == nullptr && std::get<1>(*it)->isSymbolized() == false)
-            // FIXME: Should extract nodes
-            storeAccess.erase(it++);
+        /* Clear implicit and explicit semantics - MEM */
+        this->collectUnsymbolizedNodes(storeAccess);
 
-          else if (std::get<0>(*it).getLeaAst()->isSymbolized() == false && std::get<1>(*it)->isSymbolized() == false)
-            // FIXME: Should extract nodes
-            storeAccess.erase(it++);
```

Comment:
I don't understand what was done with Lea checks here. Can you help me?
BTW, the new code looks far more readable, it is great!!